scripts/west_commands/ncs_commands.py

```diff
@@ -17,17 +17,56 @@
 
 import ncs_west_helpers as nwh
 
+def add_zephyr_rev_arg(parser):
+    parser.add_argument('-z', '--zephyr-rev',
+                        help='''upstream git ref to use for zephyr project;
+                        default is upstream/master''')
+
+def add_projects_arg(parser):
+    parser.add_argument('projects', metavar='PROJECT', nargs='*',
+                        help='''projects (by name or path) to operate on;
+                        defaults to all projects in the manifest which
+                        are shared with the upstream manifest''')
+
+def check_west_version():
+    min_ver = '0.6.99'
+    if (packaging.version.parse(west_ver) <
+            packaging.version.parse(min_ver)):
+        log.die('this command currently requires a pre-release west '
+                f'(your west version is {west_ver}, '
+                f'but must be at least {min_ver}')
```

Comment:
Yes, this could be added to core zephyr, but honestly this is generic code that is not really west-specific. west already exports its `__version__`.

Reply:
This *is* built into west... as of west's master branch :). So we can't start to rely on it until that is merged. That's what the manifest schema version feature is about.

Reply:
I think @tejlmand meant the `check_west_version()` function. That is not built into west.

Reply:
It's true that the west version can move independently of the manifest schema version, but in practice users are only going to need a minimum version of west if either the manifest schema changes or the APIs change. In this case it's because of APIs. And I don't think we'll be updating APIs without also having a change to a manifest schema in practice, since it wouldn't be necessary so I think practically this is the same thing.

But we can certainly add this to core west if we like.

Reply:
https://github.com/zephyrproject-rtos/west/issues/355

Reply:
Nice.
Only reason for commenting here, was because this was where I spotted it, and wanted to hear opinions before rasing it as an enhancement issue.